core/lib/zksync_core/src/api_server/web3/tests/mod.rs

```diff
@@ -123,10 +138,19 @@ async fn spawn_server(
 
     let server_builder = match transport {
         ApiTransportLabel::Http => ApiBuilder::jsonrpsee_backend(api_config, pool).http(0),
-        ApiTransportLabel::Ws => ApiBuilder::jsonrpc_backend(api_config, pool)
-            .ws(0)
-            .with_polling_interval(POLL_INTERVAL)
-            .with_subscriptions_limit(100),
+        ApiTransportLabel::Ws => {
+            let mut builder = ApiBuilder::jsonrpsee_backend(api_config, pool)
+                .ws(0)
+                .with_polling_interval(POLL_INTERVAL)
+                .with_subscriptions_limit(100);
```

Comment:
BTW, is `subscriptions_limit` used anywhere after refactoring? It seems like it's not, and the corresponding config param doesn't really do anything (although I have doubts it worked as intended previously). I wonder how it *should* work; unlike with filters, there's no obvious "least recently used" indicator with subscriptions, so there's no simple policy to evict "old" subscriptions when newer ones arrive.

Reply:
Brought it back for ws server via `max_connections`.

Reply:
That's what I meant by "hav[ing] doubts it worked as intended". The EN doc comment on `subscriptions_limit` reads: "Max possible limit of subscriptions to be in the API state at once", which setting the number of *connections* won't do (there may be multiple subscriptions per connection). Or am I missing something, and `max_connections` actually configures the number of subscriptions?

Reply:
No I think that those two were used interchangably bc the assumption was that statistically they are more or less equal.

Reply:
IMO, it'd make sense to rename the config param to "connections limit" so that it's less confusing. Can be done separately, of course.

Reply:
yes let's do it separately because that would lead to deployments change